Spark 3: Connection to Hive Metastore doesn't work when Hive Server 2 is running on the edge

Similar to #394 but with Spark 3.

The connection to the Hive Metastore from Spark does not work. We can see this by doing a simple `SHOW DATABASES` in `pyspark3` shell:
```py
spark.sql('SHOW DATABASES')
```
With Spark 3, we have the following error:
```
java.lang.RuntimeException: Unable to instantiate org.apache.hadoop.hive.ql.metadata.SessionHiveMetaStoreClient
...
Caused by: java.lang.IllegalArgumentException: java.io.IOException: Configuration problem with provider path.
	at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.open(HiveMetaStoreClient.java:459)
	at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.<init>(HiveMetaStoreClient.java:224)
	at org.apache.hadoop.hive.ql.metadata.SessionHiveMetaStoreClient.<init>(SessionHiveMetaStoreClient.java:94)
	... 80 more
Caused by: java.io.IOException: Configuration problem with provider path.
	at org.apache.hadoop.conf.Configuration.getPasswordFromCredentialProviders(Configuration.java:2363)
	at org.apache.hadoop.conf.Configuration.getPassword(Configuration.java:2282)
	at org.apache.hadoop.hive.metastore.conf.MetastoreConf.getPassword(MetastoreConf.java:1537)
	at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.open(HiveMetaStoreClient.java:452)
	... 82 more
Caused by: java.io.FileNotFoundException: /etc/hive/conf.s2/hive.jceks (Permission denied)
```
It seems that Spark tries to use Hive CrendentialProvider, but it shouldn't be needed. It is probably a client configuration issue.

--- a/tdp_vars_defaults/spark3/spark3.yml
+++ b/tdp_vars_defaults/spark3/spark3.yml
@@ -153,3 +153,4 @@ hive_site_spark:
   hive.metastore.kerberos.principal: "hive_ms/_HOST@{{ realm }}"
   hive.metastore.execute.setugi: "true"
   hadoop.rpc.protection: AUTHENTICATION
+  hive.execution.engine: spark

